xds/internal/xdsclient/xdsresource/unmarshal_eds_test.go

```diff
@@ -99,6 +99,31 @@ func (s) TestEDSParseRespProto(t *testing.T) {
 			}(),
 			want: EndpointsUpdate{},
 		},
+		{
+			name: "max sum of weights at the same priority exceeded",
+			m: func() *v3endpointpb.ClusterLoadAssignment {
+				clab0 := newClaBuilder("test", nil)
+				clab0.addLocality("locality-1", 1, 0, []string{"addr1:314"}, &addLocalityOptions{
+					Health: []v3corepb.HealthStatus{v3corepb.HealthStatus_UNHEALTHY},
```

Comment:
The options specified using the last argument to `addLocality` controls fields of the individual endpoints. So, the health status and the weight specified using this option controls the health status and the weight of the individual endpoints (specified using the `addrsWithPort` argument). These do not matter in any way for this test. So, we can simplify this test as follows:

```golang
m: func() *v3endpointpb.ClusterLoadAssignment {
	clab0 := newClaBuilder("test", nil)
	clab0.addLocality("locality-1", 1, 0, []string{"addr1:314"}, nil)
	clab0.addLocality("locality-2", 4294967295, 1, []string{"addr2:159"}, nil)
	return clab0.Build()
	}(),
```